src/sage/rings/cfinite_sequence.py

```diff
@@ -650,28 +649,27 @@ def __getitem__(self, key):
             m = max(key.start, key.stop)
             return [self[ii] for ii in range(*key.indices(m + 1))]
         elif isinstance(key, Integral):
-            from sage.matrix.constructor import Matrix
-            d = self._deg
-            if (self._off <= key and key < self._off + len(self._a)):
-                return self._a[key - self._off]
-            elif d == 0:
+            den = self.denominator()
+            num = self.numerator()
+            if self._off >= 0:
+                num = num.shift(-self._off)
+            else:
+                den = den.shift(self._off)
+            n = key - self._off
+            if n < 0:
                 return 0
-            (quo, rem) = self.numerator().quo_rem(self.denominator())
-            wp = quo[key - self._off]
-            if key < self._off:
-                return wp
-            A = Matrix(QQ, 1, d, self._c)
-            B = Matrix.identity(QQ, d - 1)
-            C = Matrix(QQ, d - 1, 1, 0)
-            if quo == 0:
-                off = self._off
-                V = Matrix(QQ, d, 1, self._a[:d][::-1])
+            (quo, num) = num.quo_rem(den)
+            P = self.parent().polynomial_ring()
+            x = self.parent().gen()
```

Comment:
I find it suspect that `CFiniteSequences(ZZ).gen()` returns a polynomial rather than a C-finite sequence, so I would write `x = P.gen()` here to be safe.

Reply:
Yes, I agree they are equivalent, but I view the current definition of `gen()` as a bug.